SystemManager/Inc/SM_States.hpp

```diff
@@ -0,0 +1,112 @@
+/*
+ * SM_States.hpp
+ *
+ * System Manager State-Machine header file
+ * 
+ * Created on: Nov 7, 2022
+ * Author(s): Gordon Fountain
+ *
+ * Describes the states of SM:
+ * 
+ * Boot (executes only once) -> Disarm -> Ground Ops -> Flight
+ * 
+ * Also holds fatalFailure to shut down all other systems.
+ *
+ */
+
+#ifndef ZPSW3_SM_STATES_HPP
+#define ZPSW3_SM_STATES_HPP
+
+#include "SM_StateManager.hpp"
+#include "SM_DataTypes.hpp"
+
+
+class BootMode: public SystemState {
+    public:
+        void enter(SystemManager *sys_man) {
+            (void) sys_man;
+        }
```

Comment:
I have been dying not being able to make them virtual. The ONLY reason I haven't is because I am trying to exactly match the syntax of the state system that flew and was tested 2-3 years ago.

After this MVP deadline I will make these virtual and properly override them and remove the bloat.